src/httpserver/http_request.hpp

```diff
@@ -298,7 +298,12 @@ class http_request {
       * @param value The value assumed by the argument
      **/
      void set_arg(const std::string& key, const std::string& value) {
-         cache->unescaped_args[key] = value.substr(0, content_size_limit);
+         auto existing_iter = cache->unescaped_args.find(key);
+         if (existing_iter != cache->unescaped_args.end()) {
```

Comment:
Similarly I think this is also just `cache->unescaped_args[key].push_back(value.substr(0, content_size_limit));`, and below.